lib/Runtime/Library/JavascriptArray.h

```diff
@@ -645,8 +645,7 @@ namespace Js
         template<typename T, typename P>
         static BOOL TryTemplatedGetItem(T *arr, P index, Var *element, ScriptContext *scriptContext, bool checkHasItem = true)
         {
-            AssertOrFailFastMsg(false, "FIXME");
-            return /*T::Is(arr) ? JavascriptArray::TemplatedGetItem(arr, index, element, scriptContext, checkHasItem) :*/
+            return LegacyVarIs<T>(arr) ? JavascriptArray::TemplatedGetItem(arr, index, element, scriptContext, checkHasItem) :
```

Comment:
```VarIs<T>(T* arg)``` should static_assert - that sort of call means we've already done the typecast that we're trying to check. This function should be reworked, imo, to something more like
```cpp

template<typename TGT, typename T, typename P>
static BOOL TryTemplatedGetItem(T *arr, P index, Var *element, ScriptContext *scriptContext, bool checkHasItem = true)
{
    return VarIs<TGT>(arr) ? JavascriptArray::TemplatedGetItem(UnsafeVarTo<TGT>(arr), index, element, scriptContext, checkHasItem)
                           : JavascriptOperators::GetItem(arr, index, element, scriptContext);
}
```

Reply:
Agreed, but I'm trying to keep this change a strict refactor without behavior modifications, to mitigate the risk of such a large change. Could this be deferred to a follow-up?

---
In reply to: [202229437](https://github.com/Microsoft/ChakraCore/pull/5456#discussion_r202229437) [](ancestors = 202229437)

Reply:
Sure